packages/svelte/src/internal/client/runtime.js

```diff
@@ -444,10 +443,13 @@ function flush_queued_effects(effects) {
 	try {
 		for (var i = 0; i < length; i++) {
 			var signal = effects[i];
+			var flags = signal.f;
 
-			if ((signal.f & (DESTROYED | INERT)) === 0) {
+			if ((flags & (DESTROYED | INERT)) === 0) {
 				if (check_dirtiness(signal)) {
 					execute_effect(signal);
+				} else if ((flags & MAYBE_DIRTY) !== 0) {
+					set_signal_status(signal, CLEAN);
```

Comment:
this doesn't do anything — `check_dirtiness` already sets the status to `CLEAN` if it was `MAYBE_DIRTY`

```suggestion
```